### PR TITLE
only allow deploy trusted build URL if ensure_trusted flag is true

### PR DESCRIPTION
**Pull request by yongwen**
add the schema change to schema sql. this is need for a fresh DB.
and fix the whitespace in migration script.

### CODE REVIEW

File: deploy-service/common/src/main/java/com/pinterest/deployservice/whitelists/Whitelist.java [hunk 1]
```diff
@@ -2,4 +2,5 @@
 
 public interface Whitelist {
```
Comment (krockpot) [inline deploy-service/common/src/main/java/com/pinterest/deployservice/whitelists/Whitelist.java:3]: while we're here do you think we could update whitelist->allowlist?

File: deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/EnvDeploys.java [hunk 2]
```diff
@@ -212,16 +215,24 @@ public Response create(
...
+            throw new TeletaanInternalException(Response.Status.BAD_REQUEST, 
+                String.format("Build name (%s) does not match stage config (%s).", 
+                    buildBean.getBuild_name(), envBean.getBuild_name()));
         }
+        // if the stage is not whitelisted (allow_private_build)
+        if(! envBean.getAllow_private_build()) { 
+            // only allow deploy if it is not private build
+            if (buildBean.getScm_branch().equals("private")) {
+                throw new TeletaanInternalException(Response.Status.BAD_REQUEST, 
+                    "This stage does not allow deploying a private build. Please Contact #teletraan to whitelist your stage for deploying private build");
```
Comment (krockpot) [inline deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/EnvDeploys.java:227]: whitelist->allow

File: deploy-service/common/src/main/java/com/pinterest/deployservice/whitelists/Whitelist.java (hunk 1 above)
Comment (yongwen) [reply deploy-service/common/src/main/java/com/pinterest/deployservice/whitelists/Whitelist.java:3]: good point. will do

Comment (yongwen) [reply deploy-service/common/src/main/java/com/pinterest/deployservice/whitelists/Whitelist.java:3]: decide to not do this re-factor at this time. this PR will focus on supporting the trusted URL. We can do the refactor in a later time.

File: deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/EnvDeploys.java (hunk 2 above)
Comment (yongwen) [reply deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/EnvDeploys.java:227]: see comments above.

**Review by krockpot — APPROVED** [review]
should we make “trustedBuild” instead something more concrete, like productionBuildLocations or something? in looking at the code its hard to logically understand the difference between the 2 lists passed in

**Review by ankilosaurus — APPROVED** [review]

**Pull request merged**